src/main/java/picard/vcf/GenotypeConcordance.java

```diff
@@ -301,6 +341,89 @@ private boolean indexExists(final File vcf) {
+                    }
+                    else {
+                        throw new IllegalStateException("Both truth and call contexts are null!");
+                    }
+
+                    // Add the genotypes
+                    if (truthContext != null && !truthAlleles.isEmpty()) {
+                        final Genotype truthGenotype = truthContext.getGenotype(TRUTH_SAMPLE);
+                        final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(truthGenotype);
+                        genotypeBuilder.name(outputTruthSampleName);
+                        genotypeBuilder.alleles(truthAlleles);
+                        if (!truthGenotype.hasAnyAttribute(VCFConstants.GENOTYPE_KEY)) {
+                            genotypeBuilder.attribute(VCFConstants.GENOTYPE_KEY, MISSING_VALUE_v4);
+                        }
+                        genotypes.add(genotypeBuilder.make());
+                    }
+                    else {
+                        final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(outputTruthSampleName);
+                        genotypeBuilder.alleles(Arrays.asList(Allele.NO_CALL, Allele.NO_CALL));
+                        genotypeBuilder.attribute(VCFConstants.GENOTYPE_KEY, MISSING_VALUE_v4);
+                        genotypes.add(genotypeBuilder.make());
+                    }
+                    if (callContext != null && !callAlleles.isEmpty()) {
+                        final Genotype callGenotype = callContext.getGenotype(CALL_SAMPLE);
+                        final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(callGenotype);
+                        genotypeBuilder.name(outputCallSampleName);
+                        genotypeBuilder.alleles(callAlleles);
+                        if (!callGenotype.hasAnyAttribute(VCFConstants.GENOTYPE_KEY)) {
+                            genotypeBuilder.attribute(VCFConstants.GENOTYPE_KEY, MISSING_VALUE_v4);
+                        }
+                        genotypes.add(genotypeBuilder.make());
+                    }
+                    else {
+                        final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(outputCallSampleName);
+                        genotypeBuilder.alleles(Arrays.asList(Allele.NO_CALL, Allele.NO_CALL));
+                        genotypeBuilder.attribute(VCFConstants.GENOTYPE_KEY, MISSING_VALUE_v4);
+                        genotypes.add(genotypeBuilder.make());
+                    }
+
+                    // set the alleles and genotypes
+                    builder.alleles(alleles.allAlleles).genotypes(genotypes);
+
+                    // set the concordance state attribute
+                    final TruthAndCallStates state = GenotypeConcordance.determineState(truthContext, TRUTH_SAMPLE, callContext, CALL_SAMPLE, MIN_GQ, MIN_DP);
+                    final ContingencyState[] stateArray = scheme.getConcordanceStateArray(state.truthState, state.callState);
```

Comment:
I think it would be nicer to output something other than the contingency values here because I think those are a bit confusing until you understand their purpose.  I.e. seeing something labelled `TP,FP` is a little odd.  But I don't think there's really much else that could be emitted instead that's ready to go.